pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -66,7 +70,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	}
 
 	r.log.Debug("running")
-	if !instance.Spec.OperatorFlags.GetSimpleBoolean(managed) {
+	nodeList := &corev1.NodeList{}
+	err = r.client.List(ctx, nodeList)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	removeMHCCR := len(nodeList.Items) == maxSupportedNodes
```

Comment:
This looks good to me, I think we could add one more unit test in `machinehealthcheck_controller_test.go` that confirms MHC resource is deleted at the desired node count.

Reply:
Done. Added unit test cases for the desired node count.

Reply:
could we have cases where we actually have more than 62 nodes? what I'm getting at is that to be completely sure we should maybe use `>=` instead of `==`

Reply:
+1 to use `>= 60` we could have 63 total, in the JIRA it notes 60 workers + 3 masters. In addition we should only list out worker nodes on line 75 since the aro-mhc will not take action on masters anyway.

Reply:
As per @cadenmarchese , the MHC failure is an issue only if the total nodes count reaches 62. If the node count is above that, it means some manual support is provided to handle more than 62 nodes and it's okay to keep the MHC CRs present in the cluster for that scenario.

For `>=60 and listing only worker nodes`, I have a concern that if any case a new initiated node is a non-worker and that increments the node count to 62, in that case too the MHC CRs should be deleted, am I right?